distribution/openrct2.d.ts

```diff
@@ -2507,10 +2579,26 @@ declare global {
          */
         velocity: number;
 
+        /**
+         * The current value that represents one full revolution of the entity. The
+         * rotation of yawRotation is 360/maxRotationFrames;
+         */
+        readonly maxRotationFrames: number;
+
+        /**
+         * The current rotation of the entity on the Z axis.
+         */
+        yawRotation: number;
```

Comment:
Yaw is a number because it is ordered - if you add 1 to your yaw, you will turn a specific amount every time. Roll and pitch are enums because they are out of order. You can see the order in the TrackSlope and TrackBank enum.

The normal way 3-axis rotations are described is "roll" "pitch" "yaw" but OpenRCT2 does not use this convention. So it is natural to use "roll" and "pitch" instead of "bank" and "slope." I caught the roll one but didn't catch the pitch one.

Reply:
I've changed some of the names with the goal of making the name more consistent across the codebase. Older parts of the code use bank and slope and newer parts use roll and pitch, so I've adjusted things accordingly.

Reply:
I was more a fan of the original implementation. It seems bloat to have the enum defined twice just so naming is consistent internally (but inconsistent in plugin API land).

As mentioned before though, I'm not a fan of the roll/yaw/pitch implementation internally either because it's applying the roll/yaw/pitch system to something that isn't working like that system. It's basically a lie about how it works. But that's a discussion for outside this PR.

Reply:
I have consolidated the enums.